Implemented the vector template, just the basic operations

a-moreno-a opened the pull request:
TODO: integrate in graphics api and unit test

a-moreno-a committed:
Implemented the vector template, just the basic operations

TODO: integrate in graphics api and unit test

maciekgajewski commented on course6/toyapp_ariel/displaylib/vector.h:
```diff
@@ -0,0 +1,63 @@
+    T Dot(const Vector& v) const
+    {
+        auto temp = TransformVectors<std::multiplies<T>>(v);
+        return std::accumulate(temp.begin(), temp.end(), 0, std::plus<T>());
```
Interesting, but old-school. Lambdas would be better than functors,

maciekgajewski commented on course6/toyapp_ariel/displaylib/vector.h:
```diff
@@ -0,0 +1,63 @@
+    std::array<T, size> values;
+
+    template<typename F>
+    Vector TransformVectors(const Vector& v) const
```
Take F as parameter, so that it can be deduced. Then use lambdas. Will be more convenient and flexible

maciekgajewski reviewed (changes requested):
Interesting use of algorithms, but I'd like to see unit tests for the vector. for different sizes and types

a-moreno-a committed:
Refactoring code based on code review

*Added unit test for vector
*using now lambdas instead of functors inside the vector class
*added index operators

maciekgajewski reviewed (approved):
Nice!
